q2_feature_table/_filter.py

```diff
@@ -30,18 +29,25 @@ def f(data_vector, id_, metadata):
 
 
 def _filter(table, min_frequency, max_frequency, min_nonzero, max_nonzero,
-            metadata, where, axis):
+            metadata, where, axis, exclude_ids=False):
     if min_frequency == 0 and max_frequency is None and min_nonzero == 0 and\
        max_nonzero is None and metadata is None and where is None:
         raise ValueError("No filtering was requested.")
     if metadata is None and where is not None:
         raise ValueError("Metadata must be provided if 'where' is "
                          "specified.")
+    if metadata is None and exclude_ids is True:
+        raise ValueError("Metadata must be provided if 'exclude_ids' "
+                         "is specified")
 
     if metadata is not None:
         ids_to_keep = metadata.ids(where=where)
     else:
         ids_to_keep = table.ids(axis=axis)
+    if exclude_ids is True:
+        ids_to_keep = set(table.ids(axis=axis)) - set(ids_to_keep)
+    else:
+        ids_to_keep = set(ids_to_keep)
```

Comment:
Sounds good to me! You could also drop the `else` and always cast (one line less):

```python
if exclude_ids is True:
    ids_to_keep = set(table.ids(axis=axis)) - set(ids_to_keep)
ids_to_keep = set(ids_to_keep)
```

If you went that route, you could also keep the `ids_to_keep = set(ids_to_keep)` in `_get_biom_filter_function` (less changes in the `git diff`).

Your call!